pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2041,7 +2045,11 @@ func buildCompleteFilterChain(mutable *istionetworking.MutableObjects, opts buil
 		chain := mutable.FilterChains[i]
 		opt := opts.filterChainOpts[i]
 		mutable.Listener.FilterChains[i].Metadata = opt.metadata
-		mutable.Listener.FilterChains[i].Name = opt.filterChainName
+		if len(opt.filterChainName) == 0 {
+			mutable.Listener.FilterChains[i].Name = mutable.Listener.Name + FilterChainNameInfix + uuid.New().String()
```

Comment:
sorry, I don't understand how a UUID can every be deterministic? The next time we re-generate listeners the name will be different and envoy will do a full drain on the listener, right?

Reply:
Right, sorry I misunderstood. I may instead use a hash on some fields of the filterchain as the uid. What do you think?